labs/CryptoProofs/CryptoProofs.md

```diff
@@ -216,62 +243,109 @@ Here's the breakdown of this proof:
 
 **EXERCISE** 2.2.1 The other direction
 
-Our example proof showed that `g` inverts `f` for all inputs. Does this work the other way around? Try it! If the proof fails, it will provide a *counterexample*. Use the counterexample to understand what went wrong.
+Our example proof showed that `g` inverts `f` for all inputs. Does
+this work the other way around? Try it! If the proof fails, it will
+provide a *counterexample*. Use the counterexample to understand what
+went wrong.
 
 
 **EXERCISE** 2.2.2 DES inversion
 
-Use Cryptol to prove that `DES.encrypt` and `DES.decrypt` are inverses for all possible inputs. Show both directions.
+Use Cryptol to prove that `DES.encrypt` and `DES.decrypt` are inverses
+for all possible inputs. Show both directions.
 
-*Hint*: Lambda functions can take more than one input, just like normal functions! For example: `\x y z -> x+y+z`
+*Hint*: Lambda functions can take more than one input, just like
+normal functions! For example: `\x y z -> x+y+z`
 
 *Hint*: For fastest results, use the `abc` prover.
 
 
-### 2.3 Collision Detection
+### 2.3 Proof of Injectivity
```

Comment:
Yeah, you are showing it's injective, but you're also showing that it's bijective! You get surjective for free because the range and domain are the same. Maybe worthwhile to point out? Well, this same discussion happens already in the Salsa20 lab. ???

Reply:
Implemented.